spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkBucketFunction.java

```diff
@@ -74,7 +74,7 @@ public void testSpecValues() {
         .as("Spec example: hash(2017-11-16T22:31:08) = -2047944441")
         .isEqualTo(-2047944441);
 
-    assertThat(new BucketFunction.BucketString().hash("iceberg"))
+    assertThat(new BucketFunction.BucketString().hash("iceberg".getBytes(StandardCharsets.UTF_8)))
         .as("Spec example: hash(\"iceberg\") = 1210000089")
```

Comment:
See comment above about keeping hash(String), instead of changing the existing assertion I think we should leave it as is and add a new assertion that asserts that hash("iceberg.getBytes(StandardCharsets.UTF_8)").equals(hash("iceberg"))